src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -7349,11 +7349,15 @@ internal static bool IsHomogenousArray<T>(object[] args)
                 return false;
             }
 
-            return args.All(element =>
-                            {
-                                var obj = PSObject.Base(element);
-                                return obj != null && obj.GetType().Equals(typeof(T));
-                            });
+            foreach (object element in args)
```

Comment:
Why replace LINQ `All` with a `foreach`? The former looks way clearer to me, I don't need to analyse what the loop does, the algorithm is clearly stated.

Reply:
> Why replace LINQ `All` with a `foreach`? The former looks way clearer to me, I don't need to analyse what the loop does, the algorithm is clearly stated.

It's an abstraction that isn't free unfortunately. In most places it's fine, but performance sensitive code should avoid it when possible.

Reply:
Do you have some performance measurements to show the difference? Or, more generally, does PowerShell have performance regression tests in CI?

Back at university I wrote a constraint solver that was pretty much just couple dozen expression-bodied functions immediately returning a LINQ expression. So every single line of LINQ was in hot path. I benched it against unrolled for loops and saw no statistically significant difference. And that was 5 years ago, I would imagine compiler/runtime get even better in optimizing abstractions to make them zero cost.